File: pathfinder-core/src/main/java/de/cubbossa/pathfinder/command/VisualizerTypeMessageExtension.java
package de.cubbossa.pathfinder.command;

import de.cubbossa.pathfinder.visualizer.PathVisualizer;
import de.cubbossa.tinytranslations.Message;

public interface VisualizerTypeMessageExtension<VisualizerT extends PathVisualizer<?, ?>> {

  Message getInfoMessage(VisualizerT element);
}

File: pathfinder-core/src/main/java/de/cubbossa/pathfinder/command/VisualizerTypeMessageExtension.kt
package de.cubbossa.pathfinder.command

import de.cubbossa.pathfinder.visualizer.PathVisualizer
import de.cubbossa.translations.Message

interface VisualizerTypeMessageExtension<in VisualizerT : PathVisualizer<*, *>> {
    fun getInfoMessage(element: VisualizerT): Message?
}
